src/main/java/net/backupcup/mcde/screen/handler/GildingFoundryScreenHandler.java

```diff
@@ -73,19 +99,22 @@ public int getMaxItemCount() {
 
     @Override
     public boolean onButtonClick(PlayerEntity player, int id) {
-        if (!player.isCreative()) {
-            startProgress();
+        if (generatedEnchantment.isEmpty()) {
             return false;
         }
         var weaponStack = inventory.getStack(0);
-        var gildedEnchantment = EnchantmentUtils.generateEnchantment(weaponStack);
-        if (gildedEnchantment.isEmpty()) {
+        var enchantmentId = generatedEnchantment.get();
+        if (player.isCreative()) {
+            var slots = EnchantmentSlots.fromItemStack(weaponStack);
+            slots.setGilding(enchantmentId);
```

Comment:
Consider `setGilded` verses `setGilding`. Whilst, yes, the player is gilding the item, the method is referring to the state of the item rather than the action of the user. As such, `setGilded` reflects this better.